src/index.js

```diff
@@ -38,51 +40,38 @@ addParseCloud();
 // "dotNetKey": optional key from Parse dashboard
 // "restAPIKey": optional key from Parse dashboard
 // "javascriptKey": optional key from Parse dashboard
+
 function ParseServer(args) {
-  if (!args.appId || !args.masterKey) {
-    throw 'You must provide an appId and masterKey!';
-  }
+  // Setup providers
+  CacheProvider.setup(args.cache);
+  FilesProvider.setup(args.files);
+  DatabaseProvider.setup(args.database);
+
+  // Instantiate the app
+  var app = new ParseApp(args.app);
 
-  if (args.databaseAdapter) {
-    DatabaseAdapter.setAdapter(args.databaseAdapter);
-  }
-  if (args.filesAdapter) {
-    FilesAdapter.setAdapter(args.filesAdapter);
-  }
-  if (args.databaseURI) {
-    DatabaseAdapter.setAppDatabaseURI(args.appId, args.databaseURI);
-  }
   if (args.cloud) {
+    // Add the Parse.Cloud global function definitions
     addParseCloud();
-    if (typeof args.cloud === 'function') {
-      args.cloud(Parse)
-    } else if (typeof args.cloud === 'string') {
-      require(args.cloud);
+
+    // Load the cloud code entry point
+    if (typeof args.cloud.entry === 'function') {
```

Comment:
This might be more relevant for a different PR, but I had the idea of basically passing in a cloud code configuration:

```
new ParseServer({
    cloud: {
        entry: "path" //or function
    }
```

I feel like entry (or rather, it should be entrypoint) is more semantically accurate as it's the entrypoint for the app's entire cloudcode implementation. I'm not opposed to reverting it for the sake of all the other changes though, it's just semantics after all.